ts/nni_manager/training_service/reusable/environments/heterogenousEnvironmentService.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import { EventEmitter } from "events";
+import * as fs from 'fs';
+import * as path from 'path';
+import * as component from '../../../common/component';
+import { getLogger, Logger } from '../../../common/log';
+import { TrialConfigMetadataKey } from '../../common/trialConfigMetadataKey';
+import { HeterogenousCommandChannel } from '../channels/heterogenousCommandChannel';
+import { CommandChannel } from "../commandChannel";
+import { EnvironmentInformation, EnvironmentService } from '../environment';
+import { AMLEnvironmentService } from './amlEnvironmentService';
+import { RemoteEnvironmentService } from './remoteEnvironmentService';
+import { randomSelect } from '../../../common/utils';
+
+
+/**
+ * Collector PAI jobs info from PAI cluster, and update pai job status locally
+ */
+@component.Singleton
+export class HeteroGenousEnvironmentService extends EnvironmentService {
+    
+    private amlEnvironmentService: AMLEnvironmentService;
+    private remoteEnvironmentService: RemoteEnvironmentService;
+
+    private readonly log: Logger = getLogger();
+
+    constructor() {
+        super();
+        this.amlEnvironmentService = new AMLEnvironmentService();
+        this.remoteEnvironmentService = new RemoteEnvironmentService();
+    }
+
+    public get hasStorageService(): boolean {
+        return false;
```

Comment:
It's a tricky place, doesn't work with OpenPAI. So it's better to implement some logic in trial dispatcther to support multiple platform.

Reply:
removed heterogenousEnvironmentService.ts